### PR TITLE
Add '--help' for script/configure-web-proxy

### PR DESCRIPTION
**Pull request by okurz**
Related progress issue: https://progress.opensuse.org/issues/65933

### CODE REVIEW

**Comment by okurz** [comment]
@perlpunk this is merely an experiment how far we want to go with "--help" on scripts for https://progress.opensuse.org/issues/65933 . We have handled all important main startup scripts already and removed all TODOs from t/44-scripts.t and we might want to accept this PR here but in general I would not require a full help text for all the scripts as many are single-purpose scripts which are mainly called internally. WDYT?

File: script/configure-web-proxy [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+opts=$(getopt -o h --long help -n 'parse-options' -- "$@") || usage
+eval set -- "$opts"
+while true; do
+  case "$1" in
+    -h | --help ) usage; shift ;;
+    -- ) shift; break ;;
+    * ) break ;;
+  esac
+done
+
```
Comment (okurz) [inline script/configure-web-proxy:23]: As long as we only look for a single "-h|--help" I would use the single-line approach as in `script/fetchneedles`, see below, to save some LOC, agreed?

Comment (kraih) [reply script/configure-web-proxy:23]: Personally i prefer this more verbose style for usage messages.

Comment (okurz) [reply script/configure-web-proxy:23]: hm, ok. I guess I can keep the PR as is then.

Comment (kalikiana) [reply script/configure-web-proxy:23]: This will also make it easier for someone to add a `--nginx` switch ;-)

**Comment by kalikiana** [comment]
> No test changes? That's what motivated this originally, no? Or am I missing something?

Half my comment got lost... I like the `--help` output. Happy to approve once the above q is answered :partying_face:

**Comment by okurz** [comment]
@kalikiana good that you ask. The original motivation of https://progress.opensuse.org/issues/65933 is still followed here but my main goal in the ticket and as written down as acceptance criteria was to get rid of the TODOs that perlpunk added in the test. What I did here is just some beautification on top. I could add the scripts where I added help explicitly to t/44-scripts.t if you think this is helpful. I experimented with just adding *all* files from script/* into t/44-scripts.t which does not work for all the minor important helper scripts that are missing help but also it would be dangerous because t/44-scripts.t would just call all scripts with `--help` even if the script takes no parameter at all so it could do whatever the script would do.

**Pull request merged**